### PR TITLE
DOCSP-5779 - Update mongosqld command line options and config file settings.

### PR DESCRIPTION
**Pull request by jdestefano-mongo**
JIRA: https://jira.mongodb.org/browse/DOCSP-5779

**Staging**

- `--schemaMode` : https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/reference/mongosqld.html#cmdoption-mongosqld-schemamode

- `--schemaSource` : https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/reference/mongosqld.html#cmdoption-mongosqld-schemasource

- `--schemaName` : https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/reference/mongosqld.html#cmdoption-mongosqld-schemaname

- Config file settings: https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/reference/mongosqld.html#data-sampling-options

- Schema Sampling Reference Chart: https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/reference/mongosqld.html#sampling-mode-reference-chart

- Schema Mapping Landing Page: https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/schema-configuration.html

- Pages renamed and examples updated:
  - https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/schema/cached-sampling.html
  - https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/jdestefano/DOCSP-5779/schema/persist-schema.html

### CODE REVIEW

File: source/includes/options-mongosqld.yaml [hunk 1]
```diff
@@ -101,23 +101,57 @@ description: |
...
+name: schemaName
+directive: option
+args: <db-name>
+default: "``defaultSchema``"
+description: |
+  .. versionadded:: 2.11    
+
+  Optional. The name of the schema to load from or write to
+  the :option:`--schemaSource <mongosqld --schemaSource>` database
+  depending on the value of :option:`--schemaMode <mongosqld --schemaMode>`:
```
Comment (steveren) [inline source/includes/options-mongosqld.yaml]: It's been a while since I've worked with this program, but at this point I'm having a hard time figuring this sentence out. It sounds as if the schema in "The name of the schema to load from..." is a schema file, but I think it's actually a set of definitions stored in a collection, is that correct? I think we should avoid speaking of "the schema" in the abstract; we should refer either to a text file or a collection which can hold one or more schema definitions.

File: source/includes/options-mongosqld.yaml [hunk 2]
```diff
@@ -139,47 +173,56 @@ description: |
...
 
-default: read
+  - Sample and write schema information to the MongoDB database
+    named in the :option:`--schemaSource <mongosqld --schemaSource>`
+    option.
+
+  - Read previously sampled schema information to the MongoDB
+    database named in the
+    :option:`--schemaSource <mongosqld --schemaSource>` option.
+
```
Comment (steveren) [inline source/includes/options-mongosqld.yaml]: I see that the options are "custom" and "auto", but it's not clear which option does what. I'd be explicit about the behavior of both options here.

File: source/includes/options-mongosqld.yaml [hunk 3]
```diff
@@ -101,23 +101,57 @@ description: |
...
+       - ``--schemaName`` Behavior
+
+     * - ``custom``
+       - The name of the schema to load from the database specified
+         by the :option:`--schemaSource <mongosqld --schemaSource>`
+         option.
+
+     * - ``auto``
+       - The name of the schema to write to the :option:`--schemaSource
+         <mongosqld --schemaSource>`
```
Comment (steveren) [inline source/includes/options-mongosqld.yaml]: I think we need to flesh out what's happening here a little more. If I'm understanding this correctly, "--schemaName foo --schemaMode auto" will sample a mongodb database and store the schema information in a new collection called foo. Is that right? Either way we could be more explicit about it. The Sampling Mode Reference Chart is helpful, but it doesn't include the new --schemaName option.

File: source/includes/options-mongosqld.yaml (hunk 2 above)
Comment (jdestefano-mongo) [reply source/includes/options-mongosqld.yaml]: Yeah, unless I'm missing something from the design doc, renaming these values only makes this whole thing more confusing.

Comment (jdestefano-mongo) [reply source/includes/options-mongosqld.yaml]: Fair point though, took a stab at clarifying this.

File: source/includes/options-mongosqld.yaml (hunk 1 above)
Comment (jdestefano-mongo) [reply source/includes/options-mongosqld.yaml]: The schema file is from mongodrdl and isn't covered here.

You could start up a BIC and have it sample and write the data:
```
mongosqld --schemaMode auto --schemaSource foo --schemaName bar
```

Let's say the sampling was good and things worked well, you could load that sampled data next time and have the BIC not sample or overwrite anything:
```
mongosqld --schemaMode custom --schemaSource foo --schemaName bar
```

I can't think of any other way around not using "schema" unless we go with "schema data" or "schema information"

Comment (jdestefano-mongo) [reply source/includes/options-mongosqld.yaml]: or "schema mapping"

File: source/includes/options-mongosqld.yaml (hunk 3 above)
Comment (jdestefano-mongo) [reply source/includes/options-mongosqld.yaml]: Being that these are so intertwined, I think we really need to defer to the chart. I did tighten this section up a bit.

Let me know if there are any other ways I can clarify.

File: source/includes/options-mongosqld.yaml (hunk 1 above)
Comment (steveren) [reply source/includes/options-mongosqld.yaml]: I'd go with "schema data".

File: source/reference/mongosqld.txt [hunk 4]
```diff
@@ -1023,25 +1031,27 @@ with each.
      - ``mongosqld`` Options
      - Behavior
 
-   * - Standalone Reader
-     - - :option:`--sampleMode <mongosqld --sampleMode>` ``read``
-       - :option:`--sampleSource <mongosqld --sampleSource>` *not used*
+   * - Standalone Schema
+     - - :option:`--schemaMode <mongosqld --schemaMode>` ``custom``
```
Comment (rychipman) [inline source/reference/mongosqld.txt]: `--schemaMode` should be unused for standalone

File: source/reference/mongosqld.txt [hunk 5]
```diff
@@ -1023,25 +1031,27 @@ with each.
...
+       - :option:`--schemaSource <mongosqld --schemaSource>` *not used*
      - :manual:`Standalone </reference/glossary/#term-standalone>`
        :program:`mongod` instance. :binary:`~bin.mongosqld`
```
Comment (rychipman) [inline source/reference/mongosqld.txt]: Not something you changed this time around, but this is actually unrelated to the MongoDB topology (perhaps "standalone" was a poor choice of terminology)

File: source/reference/mongosqld.txt [hunk 6]
```diff
@@ -1023,25 +1031,27 @@ with each.
...
-   * - Clustered Reader
-     - - :option:`--sampleMode <mongosqld --sampleMode>` ``read``
-       - :option:`--sampleSource <mongosqld --sampleSource>` <dbName>
-       - :option:`--sampleRefreshIntervalSecs <mongosqld --sampleRefreshIntervalSecs>` *not used*
+   * - Custom Schema
+     - - :option:`--schemaMode <mongosqld --schemaMode>` ``custom``
+       - :option:`--schemaSource <mongosqld --schemaSource>` <dbName>
+       - :option:`--schemaRefreshIntervalSecs <mongosqld --schemaRefreshIntervalSecs>` *not used*
      - In a :manual:`replica set </replication/>` or
        :manual:`sharded cluster </sharding>`
```
Comment (rychipman) [inline source/reference/mongosqld.txt]: as noted above, these modes don't depend on MongoDB cluster type

File: source/reference/mongosqld.txt [hunk 7]
```diff
@@ -1052,23 +1062,25 @@ with each.
...
 
-   * - Clustered Writer
-     - - :option:`--sampleMode <mongosqld --sampleMode>` ``write``
-       - :option:`--sampleSource <mongosqld --sampleSource>` <dbName>
-       - :option:`--sampleRefreshIntervalSecs  <mongosqld --sampleRefreshIntervalSecs>` > 0
+   * - Auto Schema
+     - - :option:`--schemaMode <mongosqld --schemaMode>` ``auto``
+       - :option:`--schemaSource <mongosqld --schemaSource>` <dbName>
+       - :option:`--schemaRefreshIntervalSecs  <mongosqld --schemaRefreshIntervalSecs>` > 0
      - Read and persist schema data in a user-specified schema
```
Comment (rychipman) [inline source/reference/mongosqld.txt]:
```suggestion
     - Sample and persist schema data in a user-specified schema
```
Just a suggestion, so we're explicit about the distinction between custom and auto modes

File: source/reference/mongosqld.txt [hunk 8]
```diff
@@ -1023,25 +1031,27 @@ with each.
...
-     - - :option:`--sampleMode <mongosqld --sampleMode>` ``read``
-       - :option:`--sampleSource <mongosqld --sampleSource>` <dbName>
-       - :option:`--sampleRefreshIntervalSecs <mongosqld --sampleRefreshIntervalSecs>` *not used*
+   * - Custom Schema
+     - - :option:`--schemaMode <mongosqld --schemaMode>` ``custom``
+       - :option:`--schemaSource <mongosqld --schemaSource>` <dbName>
+       - :option:`--schemaRefreshIntervalSecs <mongosqld --schemaRefreshIntervalSecs>` *not used*
      - In a :manual:`replica set </replication/>` or
        :manual:`sharded cluster </sharding>`
        environment, sample data only at startup. Read schema data
```
Comment (rychipman) [inline source/reference/mongosqld.txt]: Custom mode doesn't sample at all, but rather reads a stored schema from the `schemaSource` db

**Comment by jdestefano-mongo** [comment]
@rychipman Ping for a follow-up review when you have some bandwidth
